app/models/concerns/taggable.rb

```diff
@@ -0,0 +1,43 @@
+module Taggable
+  extend ActiveSupport::Concern
+
+  included do
+    class_attribute :tag_column
+    self.tag_column = :tags
+  end
+
+  class_methods do
+    def acts_as_taggable_array_on(*tag_def)
+      self.tag_column = tag_def.first
+    end
+
+    def with_all_tags(tags)
+      where("array_lowercase(#{tag_column}) @> ARRAY[?]::varchar[]", downcase_tags(tags))
+    end
+
+    def with_tag(tag)
+      where("'#{tag.downcase}' = ANY (array_lowercase(#{tag_column}))")
```

Comment:
While addressing @pixeltrix's comment on my PR (https://github.com/alphagov/e-petitions/pull/603), which extends this module, I've run `EXPLAIN` on this query and found that it used a sequential scan rather than the index.

I found https://stackoverflow.com/questions/22255503/why-isnt-my-postgresql-array-index-getting-used-rails-4/22265078 and found a way to change this to use the index: 

```
    def with_tag(tag)
      where("'{#{tag}}' <@ (array_lowercase(tags))")
    end
```